pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/TimeSegmentPruner.java

```diff
@@ -18,6 +18,7 @@
  */
 package org.apache.pinot.broker.routing.segmentpruner;
 
+import com.google.common.base.Preconditions;
```

Comment:
this whole file is a refactor. you're getting the format from the `DateTimeFieldSpec` rather then unnecessarily passing it. but nothing is actually changing here.

Reply:
Correct. I started with making changes in this file, but then in the end find I can simply override the `format`